From d7ecb6cfd0d28e7456b80dfa960900d04df96141 Mon Sep 17 00:00:00 2001
Message-Id: <d7ecb6cfd0d28e7456b80dfa960900d04df96141.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Sun, 11 Jan 2015 21:42:59 +0530
Subject: [PATCH 37/95] MUST_REBASE [VPG]: video/adf/intel: Program the ports
 bit mask for dual link

Use the ports bit mask for dual link specific programming.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-5276
Change-Id: I5543405eb03c5011e8661fc5e983506bba77a0bc
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 .../core/common/dsi/panels/intel_dsi_panel_vbt.c   |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c b/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c
index f43aefe..e4548dd 100644
--- a/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c
+++ b/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c
@@ -649,6 +649,9 @@ static int panel_generic_init(struct dsi_pipe *pipe)
 		mipi_config->bta_enabled ? DISABLE_VIDEO_BTA : 0;
 	intel_dsi->dual_link = mipi_config->dual_link;
 
+	if (intel_dsi->dual_link)
+		intel_dsi->ports = ((1 << PORT_A) | (1 << PORT_C));
+
 	pclk = mode->clock;
 
 	/* Burst Mode Ratio
-- 
1.7.9.5

